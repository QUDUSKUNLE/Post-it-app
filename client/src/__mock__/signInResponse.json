{
    "message": "User Signed in successfully",
    "response": [{
        "-KwW2TEuvqwcF2ODZH1-": {
            "phone_Number": "07031187445",
            "time": "Sun, Oct 15, 2017 6:21 PM",
            "userEmail": "quduskunle@gmail.com",
            "userId": "NCaAzr0ZzqfCLtXQlQG0jW2DWbg1",
            "userName": "kunle"
        }
    }]
}