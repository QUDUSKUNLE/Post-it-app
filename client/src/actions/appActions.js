// import axios from 'axios';


// /**
//  * @function addMember
//  * @param {object} member - { member }
//  * @returns {object} object
//  */
// export const addMember = (member) => axios.post('/group/member', member);
