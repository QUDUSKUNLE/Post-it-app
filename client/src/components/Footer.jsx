import React from 'react';

/**
 * @export
 * @description - renders Footer Component
 * @class Footer
 * @extends {React.Component}
 */
export default class Footer extends React.Component {
  render() {
    return (
      <div className="footer">
        <div className="container">
          <center>
           Andela 2017
          </center>
        </div>
      </div>
    );
  }
}
