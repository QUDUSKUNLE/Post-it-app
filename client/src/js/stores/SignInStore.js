import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher.js';
import {
  SIGN_IN_SUCCESS, SIGN_IN_ERROR
} from '../constants/ActionConstants.js';

/**
 * Signout Store, it hold user's state, listen to signin Actions
 * @class SignInStore
 */
class SignInStore extends EventEmitter {
  constructor(props) {
    super(props);
    this.handleActions = this.handleActions.bind(this);
  }

  handleActions(action) {
    switch (action.type) {
      case SIGN_IN_SUCCESS:
        this.emit('SIGN_OUT_SUCCESS');
        break;

      case SIGN_IN_ERROR:
        this.emit('SIGN_IN_ERROR');
        break;

      default:
    }
  }
}

// Initiate an instance of SignInStore
const signInStore = new SignInStore();

// Register a dispatcher and bind it to handleActions method
AppDispatcher.register(signInStore.handleActions.bind(signInStore));

// export an instance of SignInStore
export default signInStore;
