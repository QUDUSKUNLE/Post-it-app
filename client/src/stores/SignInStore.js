import { EventEmitter } from 'events';
import AppDispatcher from '../dispatcher/AppDispatcher';
import {
  SIGN_IN_SUCCESS,
  GOOGLE_SIGN_IN_SUCCESS, PASSWORD_RESET_SUCCESS
} from '../constants/ActionConstants';

/**
 * Signout Store, it hold user's state, listen to signin Actions
 * @class SignInStore
 */
class SignInStore extends EventEmitter {
  /**
   * @constructor
   */
  constructor() {
    super();
    this.signInMessage = {};
    this.googleSignInMessage = {};
    this.passwordResetResponse = {};
    this.signInUser = this.signInUser.bind(this);
    this.googleSignIn = this.googleSignIn.bind(this);
    this.passwordReset = this.passwordReset.bind(this);
    this.handleActions = this.handleActions.bind(this);
  }

  signInUser() {
    return this.signInMessage;
  }

  googleSignIn() {
    return this.googleSignInMessage;
  }

  passwordReset() {
    return this.passwordResetResponse;
  }

  /**
   * Receives actions and update the stores accordingly
   * @method handleActions
   * @param {object} action - Action type and data
   * @return {*} -
   */
  handleActions(action) {
    switch (action.type) {
      case SIGN_IN_SUCCESS:
        this.signInMessage = action.response;
        this.emit('SIGN_IN_SUCCESS');
        break;

      case GOOGLE_SIGN_IN_SUCCESS:
        this.googleSignInMessage = action.response;
        this.emit('GOOGLE_SIGN_IN_SUCCESS');
        break;

      case PASSWORD_RESET_SUCCESS:
        this.passwordResetResponse = action.response;
        this.emit('PASSWORD_RESET_SUCCESS');
        break;
      default:
    }
  }
}

// Initiate an instance of SignInStore
const signInStore = new SignInStore();

// Register a dispatcher and bind it to handleActions method
AppDispatcher.register(signInStore.handleActions.bind(signInStore));

// export an instance of SignInStore
export default signInStore;
