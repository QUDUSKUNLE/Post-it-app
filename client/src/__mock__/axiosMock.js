import groupMemberResponse from './groupMemberResponse.json';
import groupMessageResponse from './groupMessageResponse.json';
import groupResponse from './groupResponse.json';
import memberResponse from './memberResponse.json';
import registeredUsers from './registeredUsers.json';
import sendMessageResponse from './sendMessageResponse.json';
import signInResponse from './signInResponse.json';
import signUpResponse from './signUpResponse.json';
const passwordResetResponse = {
  message: 'Password reset email sent successfully!'
};

const signOutResponse = {
  message: 'User`s signed-out successfully'
};

const createGroupResponse = {
  message: 'Group created successfully'
};

const mockApiCall = {
  get(url) {
    if (url === '/api/v1/getgroups/userId') {
      return Promise.resolve(groupResponse);
    } else if (url === '/api/v1/getMembers/groupId') {
      return Promise.resolve(groupMemberResponse);
    } else if (url === '/api/v1/getMessage/groupId') {
      return Promise.resolve(groupMessageResponse);
    } else if (url === '/api/v1/getAllRegisteredUsers') {
      return Promise.resolve(registeredUsers);
    }
  },
  post(url) {
    if (url === '/api/v1/signup') {
      return Promise.resolve(signUpResponse);
    } else if (url === '/api/v1/signin') {
      return Promise.resolve(signInResponse);
    } else if (url === '/api/v1/passwordReset') {
      return Promise.resolve(passwordResetResponse);
    } else if (url === '/api/v1/createGroup') {
      return Promise.resolve(createGroupResponse);
    } else if (url === '/api/v1/signout') {
      return Promise.resolve(signOutResponse);
    } else if (url === '/api/v1/addmember/groupId') {
      return Promise.resolve(memberResponse);
    } else if (url === '/api/v1/sendMessage/groupId') {
      return Promise.resolve(sendMessageResponse);
    }
  }
};
export default mockApiCall;
