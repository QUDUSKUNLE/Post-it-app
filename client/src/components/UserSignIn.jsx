import React from 'react';
import GoogleButton from 'react-google-button';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import { Link } from 'react-router-dom';
import firebase from '../vendors/index';
import SignInStore from '../stores/SignInStore';
import { signInAction, signInWithGoogle } from '../actions/signInAction';

/**
 * @description - renders UserSignIn Component
 * @class UserSignIn
 * @extends {React.Component}
 */
export default class UserSignIn extends React.Component {
  /**
   * Create a constructor
   * @constructor
   * @param {object} props -
   */
  constructor(props) {
    super(props);
    this.state = {
      userName: '',
      userId: '',
      email: '',
      password: '',
      loggedIn: false,
      isLoading: false,
      signInResponse: ''
    };

    /**
     * @description This binding is necessary to make `this` work
     * in the callback
     */
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.handleSignInAction = this.handleSignInAction.bind(this);
    this.googleSignIn = this.googleSignIn.bind(this);
    this.handleGoogleEvent = this.handleGoogleEvent.bind(this);
  }

   /**
   * @method componentDidMount
   * @description Adds an event Listener to the Store and fires
   * when the component is fully mounted.
   * @return { void} void
   * @memberof UserSignIn
  */
  componentDidMount() {
    SignInStore.on('SIGN_IN_SUCCESS', this.handleSignInAction);
    SignInStore.on('GOOGLE_SIGN_IN_SUCCESS', this.handleGoogleEvent);
  }

  /**
   * @method componentWillUnmount
   * @description remove event Listener from the Store and fires.
   * @return {void} void
   * @memberof UserSignIn
   */
  componentWillUnmount() {
    SignInStore.removeListener('SIGN_IN_SUCCESS', this.handleSignInAction);
    SignInStore.removeListener('GOOGLE_SIGN_IN_SUCCESS',
      this.handleGoogleEvent);
  }

  /**
   * onChange event
   * @param {object} event - event
   * @return {void} updated state of user
   */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
   * @description - this handles SignIn form submission
   * @param {object} event - event.
   * @returns {void} .
   */
  onSubmit(event) {
    event.preventDefault();
    signInAction({ ...this.state });
  }

  /**
   * @description This handles handleSignInAction
   * @param {object} user .
   * @returns {void} .
   */
  handleSignInAction() {
    const response = SignInStore.signInUser();
    this.setState({
      loggedIn: true,
    });
    toastr.success(response.message);
    localStorage.setItem('userIn', JSON.stringify(this.state.loggedIn));
    this.props.history.push('/broadcastboard');
  }

  /**
   * @description - this handles Google SignIn Method
   * @param {object} event - event.
   * @returns {void} .
  */
  googleSignIn() {
    const provider = new firebase.auth.GoogleAuthProvider();
    provider.addScope('https://www.googleapis.com/auth/plus.login');
    firebase.auth().signInWithPopup(provider)
      .then((result) => {
        signInWithGoogle(result);
        this.setState({
          isLoading: true
        });
      }).catch(error => {
        toastr.error(error.message);
      });
  }

  /**
   * @memberof UserSignIn
   * @return {*} void
   */
  handleGoogleEvent() {
    const googleResponse = SignInStore.googleSignIn();
    this.setState({
      loggedIn: true,
    });
    toastr.success(googleResponse.message);
    localStorage.setItem('userIn', JSON.stringify(this.state.loggedIn));
    this.props.history.push('/broadcastboard');
  }

  /**
   * @description - render method, React lifecycle method
   * @returns {Object} SignIn component
   * @SignIn
   */
  render() {
    const isLoading = () => {
      const loading = (
        this.state.isLoading ? <div id="loader"></div> : <span></span>
      );
      return loading;
    };
    return (
      <div className="container signin">
        {isLoading()}
        <div className="row">
          <div className="col-md-6 col-md-offset-3">
            <h5 className="text-center">
              <b>Sign in to PostIt</b>
            </h5>
            <form onSubmit={this.onSubmit}
              className="w3-card w3-white" id="signinForm">
              <div className="form-group">
                <label htmlFor="email">Email</label>
                <input value={this.state.email} onChange={this.onChange}
                  id="email" type="email"
                  className="signinform" placeholder="johndoe@example.com"
                  name="email" required />
              </div>
              <div className="form-group">
                <div className="row">
                  <div className="col-md-4">
                    <label htmlFor="password">Password</label>
                  </div>
                </div>
                <input id="password" type="password"
                  value={this.state.password} onChange={this.onChange}
                  className="signinform" placeholder="*********"
                  name="password" required />
              </div>
              <button id="submitButton"
                type="submit" className="signinformbtn">Sign in
              </button>
            </form>
            <br/>
            <span className="pull-right create">
              <Link to="/passwordreset">
                Forgot password?
              </Link></span>
            <br/>
            <center>
              <GoogleButton onClick={this.googleSignIn} />
            </center>
            <br/>
            <div>
              <center>
                <p>New to PostIt? <Link to="/" className="create">
                  Create an account.</Link>
                </p>
              </center>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

// props validation
UserSignIn.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  })
};
