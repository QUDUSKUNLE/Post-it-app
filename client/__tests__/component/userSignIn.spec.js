import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import GoogleButton from 'react-google-button';
import { shallow, mount } from 'enzyme';
import expect from 'expect';
import sinon from 'sinon';
import 'babel-polyfill';
import firebase from '../../src/vendors/index';
import localStorageMock from '../../src/__mock__/localStorage';
import axiosMock from '../../src/__mock__/axiosMock.jsx';
import Footer from '../../src/components/Footer';
import { signInAction } from '../../src/actions/SignInActions';
import UserSignIn from '../../src/components/UserSignIn';
import SignInStore from '../../src/stores/SignInStore';
import signInResponse from '../../src/__mock__/signInResponse.json';


window.localStorage = localStorageMock;
// jest.mock('../../src/vendors/index.js', () => {
//   /**
//    * @description describes a function that mocks firebase module,
//    * fires it action to make an Api call, returns a promise that is mocked
//    *
//    * @param { void } void takes no parameter
//    *
//    * @return { object } mockfirebase object
//    *
//    * @function Test
//    */
//   function Test() {

//   }
//   const mockFirebase = jest.fn().mockReturnValue({
//     signInWithPopup: jest.fn().mockReturnValue(Promise.resolve({
//       user: 'testUser'
//     }))
//   });
//   Test.prototype.addScope = jest.fn();
//   mockFirebase.GoogleAuthProvider = Test;
//   return {
//     auth: mockFirebase,
//   };
// });

describe('UserSignIn component', () => {
  let component;
  let wrapper;
  const mockOnSignIn = sinon.stub(SignInStore,
    'on').callsFake((user, cb) => cb());
  const mockUnMountSignIn = sinon.stub(SignInStore,
    'removeListener').callsFake((user, cb) => cb());
  const mockSignInResponse = sinon.stub(SignInStore,
    'signInUser').returns(signInResponse);
  // let axiosMock;
  // const onSubmitSpy = sinon.spy();
  // const signInActionSpy = signInAction;
  beforeEach(() => {
    // axiosMock = mockApiCall;
    // jest.mock('axios', () => axiosMock);
    // SignInStore.signInUser = jest.fn(() => signInResponse);
    const props = {
      userName: '',
      userId: '',
      email: '',
      password: '',
      loggeddIn: false,
      isLoading: false
    };
    component = shallow(<UserSignIn {...props}/>);
    wrapper = mount(<UserSignIn {...props}/>,
      {
        childContextTypes: { router: PropTypes.object },
        context: {
          router: {
            history: {
              push: () => '/signin',
              replace: () => null,
              createHref: () => null,
              path: '/signin',
              component: '[function UserSignIn]',
              location: {
                pathname: '/signin',
                search: '',
                hash: '',
                key: 'on2bj3'
              },
              computedMatch: {
                path: '/signin',
                url: '/signin',
                isExact: true,
                params: {}
              }
            }
          }
        }
      }
    );
  });
  it('expects UserSignUp component to be defined', () => {
    expect(UserSignIn).toBeDefined();
  });
  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
    expect(component.find('div').length).toBe(15);
  });
  // it('should have an empty initial state', () => {
  //   console.log(component);
  //   // expect(wrapper.state()).toBeDefined();
  //   // expect(wrapper.state().userName).toEqual('');
  //   // expect(wrapper.state().userId).toEqual('');
  //   // expect(wrapper.state().email).toEqual('');
  //   // expect(wrapper.state().password).toEqual('');
  // });

  // it('should contain <Footer/>', () => {
  //   expect(wrapper.find(Footer).root.length).toEqual(1);
  //   expect(wrapper.find('form').length).toEqual(1);
  // });

  // it('should contain defined methods', () => {
  //   expect(wrapper.nodes[0].onChange).toBeDefined();
  //   expect(wrapper.nodes[0].onSubmit).toBeDefined();
  //   expect(wrapper.nodes[0].handleSignInAction).toBeDefined();
  //   expect(wrapper.nodes[0].googleSignIn).toBeDefined();
  //   expect(wrapper.nodes[0].handleGoogleEvent).toBeDefined();
  // });

  // it('It should call onChange method', () => {
  //   // const event = { target: { name: 'name', value: 'value' } };
  //   // wrapper.instance().onChange(event);
  //   // expect(wrapper.state().name).toEqual('value');
  // });
  // it('should call onSubmit method', () => {
  //   wrapper.instance().onSubmit({ preventDefault() {} });
  //   expect(wrapper.state().email).toEqual('');
  // });
  // it('should componentDidMount component lifecycle', () => {
  //   const spy = sinon.spy(UserSignIn.prototype, 'componentDidMount');
  //   wrapper.instance().componentDidMount();
  //   expect(spy.calledOnce).toBeTruthy();
  // });

  // it('should call componentWillUnmount component lifecycle', () => {
  //   const spy = sinon.spy(UserSignIn.prototype, 'componentWillUnmount');
  //   wrapper.instance().componentWillUnmount();
  //   expect(spy.calledOnce).toBeTruthy();
  // });
  // it('should find a link', () => {
  //   console.log(wrapper);
  //   // expect(wrapper.find(Link).at(2).prop('to')).toEqual('/signin');
  //   // expect(wrapper.find(Link).at(3).prop('to')).toEqual('/passwordreset');
  //   // expect(wrapper.find(Link).at(4).prop('to')).toEqual('/');
  // });

  // it('should render correctly', () => {
  //   expect(component).toMatchSnapshot();
  //   expect(component.find('form')).toHaveLength(1);
  //   expect(component.find('input')).toHaveLength(2);
  //   expect(component.find('input.signinform')).toHaveLength(2);
  // });
  // it('should contain a google button', () => {
  //   wrapper.find(GoogleButton).simulate('click');
  //   expect(firebase.auth.GoogleAuthProvider.prototype.addScope)
  //     .toHaveBeenCalledTimes(1);
  //   expect(firebase.auth().signInWithPopup).toHaveBeenCalled();
  // });
  // it('should take props', () => {
  //   expect(wrapper.props().onChange).toBeTruthy();
  //   expect(wrapper.props().onClick).toBeTruthy();
  //   expect(wrapper.props().handleSignInAction).toBeTruthy();
  //   expect(wrapper.props().handleGoogleEvent).toBeTruthy();
  // });
});


// describe('<UserSignIn />', () => {
//   const signInActionSpy = signInAction;
//   it('simulates click events', () => {
//     const onSubmitSpy = sinon.spy();
//     const component = shallow(<UserSignIn onSubmitSpy={signInActionSpy}/>);
//     component.find('#signinformbtn').simulate('click');
//     expect(component.props.onSubmitSpy).toHaveBeenCalled();
//   });
// });

