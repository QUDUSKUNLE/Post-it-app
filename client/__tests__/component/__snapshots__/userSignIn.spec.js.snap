// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSignIn component should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <nav
      className="navbar navbar-inverse navabar-fixed-top"
      role="navigation"
    >
      <div
        className="container-fluid"
      >
        <div
          className="navbar-header"
        >
          <button
            className="navbar-toggle collapsed"
            data-target=".navbar-collapse"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="sr-only"
            >
              Toggle navigation
            </span>
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
          </button>
          <Link
            className="navbar-brand"
            replace={false}
            to="#"
          >
            PostIt
            <small>
              App
            </small>
          </Link>
        </div>
        <div
          className="collapse navbar-collapse"
        >
          <ul
            className="nav navbar-nav navbar-right"
          >
            <li>
              <Link
                replace={false}
                to="/"
              >
                Home
              </Link>
            </li>
            <li
              className="active"
            >
              <Link
                replace={false}
                to="/signin"
              >
                Sign in
              </Link>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      className="container signin"
    >
      <span />
      <div
        className="row"
      >
        <div
          className="col-md-6 col-md-offset-3"
        >
          <div
            className="row"
          >
            <center>
              <GoogleButton
                disabled={false}
                label="Sign in with Google"
                onClick={[Function]}
                type="dark"
              />
            </center>
            <br />
            <div
              className="text-center or"
            >
              <b>
                OR
              </b>
            </div>
            <form
              className="w3-card w3-white"
              id="signinForm"
              onSubmit={[Function]}
            >
              <div
                className="form-group"
              >
                <label
                  htmlFor="email"
                >
                  Email
                </label>
                <input
                  className="signinform"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  placeholder="johndoe@example.com"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
              <div
                className="form-group"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-md-4"
                  >
                    <label
                      htmlFor="password"
                    >
                      Password
                    </label>
                  </div>
                  <div
                    className="col-md-4 col-md-offset-4"
                  >
                    <Link
                      replace={false}
                      to="/passwordreset"
                    >
                      <h6
                        className="pull-right create"
                      >
                        <b>
                          Forgot password?
                        </b>
                      </h6>
                    </Link>
                  </div>
                </div>
                <input
                  className="signinform"
                  id="password"
                  name="password"
                  onChange={[Function]}
                  placeholder="*********"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
              <button
                className="signinformbtn"
                id="submitButton"
                type="submit"
              >
                Sign in
              </button>
            </form>
          </div>
          <br />
          <div>
            <center>
              <p>
                New to PostIt App? 
                <Link
                  className="create"
                  replace={false}
                  to="/"
                >
                  Create an account.
                </Link>
              </p>
            </center>
          </div>
        </div>
      </div>
    </div>
    <Footer />
  </div>,
  "nodes": Array [
    <div>
      <nav
        className="navbar navbar-inverse navabar-fixed-top"
        role="navigation"
      >
        <div
          className="container-fluid"
        >
          <div
            className="navbar-header"
          >
            <button
              className="navbar-toggle collapsed"
              data-target=".navbar-collapse"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="sr-only"
              >
                Toggle navigation
              </span>
              <span
                className="icon-bar"
              />
              <span
                className="icon-bar"
              />
              <span
                className="icon-bar"
              />
            </button>
            <Link
              className="navbar-brand"
              replace={false}
              to="#"
            >
              PostIt
              <small>
                App
              </small>
            </Link>
          </div>
          <div
            className="collapse navbar-collapse"
          >
            <ul
              className="nav navbar-nav navbar-right"
            >
              <li>
                <Link
                  replace={false}
                  to="/"
                >
                  Home
                </Link>
              </li>
              <li
                className="active"
              >
                <Link
                  replace={false}
                  to="/signin"
                >
                  Sign in
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <div
        className="container signin"
      >
        <span />
        <div
          className="row"
        >
          <div
            className="col-md-6 col-md-offset-3"
          >
            <div
              className="row"
            >
              <center>
                <GoogleButton
                  disabled={false}
                  label="Sign in with Google"
                  onClick={[Function]}
                  type="dark"
                />
              </center>
              <br />
              <div
                className="text-center or"
              >
                <b>
                  OR
                </b>
              </div>
              <form
                className="w3-card w3-white"
                id="signinForm"
                onSubmit={[Function]}
              >
                <div
                  className="form-group"
                >
                  <label
                    htmlFor="email"
                  >
                    Email
                  </label>
                  <input
                    className="signinform"
                    id="email"
                    name="email"
                    onChange={[Function]}
                    placeholder="johndoe@example.com"
                    required={true}
                    type="email"
                    value=""
                  />
                </div>
                <div
                  className="form-group"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-md-4"
                    >
                      <label
                        htmlFor="password"
                      >
                        Password
                      </label>
                    </div>
                    <div
                      className="col-md-4 col-md-offset-4"
                    >
                      <Link
                        replace={false}
                        to="/passwordreset"
                      >
                        <h6
                          className="pull-right create"
                        >
                          <b>
                            Forgot password?
                          </b>
                        </h6>
                      </Link>
                    </div>
                  </div>
                  <input
                    className="signinform"
                    id="password"
                    name="password"
                    onChange={[Function]}
                    placeholder="*********"
                    required={true}
                    type="password"
                    value=""
                  />
                </div>
                <button
                  className="signinformbtn"
                  id="submitButton"
                  type="submit"
                >
                  Sign in
                </button>
              </form>
            </div>
            <br />
            <div>
              <center>
                <p>
                  New to PostIt App? 
                  <Link
                    className="create"
                    replace={false}
                    to="/"
                  >
                    Create an account.
                  </Link>
                </p>
              </center>
            </div>
          </div>
        </div>
      </div>
      <Footer />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <UserSignIn />,
      "_debugID": 892,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": UserSignIn {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "googleSignIn": [Function],
        "handleGoogleEvent": [Function],
        "handleSignInAction": [Function],
        "onChange": [Function],
        "onSubmit": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "email": "",
          "isLoading": false,
          "loggedIn": false,
          "password": "",
          "userId": "",
          "userName": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <nav
            className="navbar navbar-inverse navabar-fixed-top"
            role="navigation"
          >
            <div
              className="container-fluid"
            >
              <div
                className="navbar-header"
              >
                <button
                  className="navbar-toggle collapsed"
                  data-target=".navbar-collapse"
                  data-toggle="collapse"
                  type="button"
                >
                  <span
                    className="sr-only"
                  >
                    Toggle navigation
                  </span>
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                </button>
                <Link
                  className="navbar-brand"
                  replace={false}
                  to="#"
                >
                  PostIt
                  <small>
                    App
                  </small>
                </Link>
              </div>
              <div
                className="collapse navbar-collapse"
              >
                <ul
                  className="nav navbar-nav navbar-right"
                >
                  <li>
                    <Link
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>
                  </li>
                  <li
                    className="active"
                  >
                    <Link
                      replace={false}
                      to="/signin"
                    >
                      Sign in
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
          <div
            className="container signin"
          >
            <span />
            <div
              className="row"
            >
              <div
                className="col-md-6 col-md-offset-3"
              >
                <div
                  className="row"
                >
                  <center>
                    <GoogleButton
                      disabled={false}
                      label="Sign in with Google"
                      onClick={[Function]}
                      type="dark"
                    />
                  </center>
                  <br />
                  <div
                    className="text-center or"
                  >
                    <b>
                      OR
                    </b>
                  </div>
                  <form
                    className="w3-card w3-white"
                    id="signinForm"
                    onSubmit={[Function]}
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="email"
                      >
                        Email
                      </label>
                      <input
                        className="signinform"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="johndoe@example.com"
                        required={true}
                        type="email"
                        value=""
                      />
                    </div>
                    <div
                      className="form-group"
                    >
                      <div
                        className="row"
                      >
                        <div
                          className="col-md-4"
                        >
                          <label
                            htmlFor="password"
                          >
                            Password
                          </label>
                        </div>
                        <div
                          className="col-md-4 col-md-offset-4"
                        >
                          <Link
                            replace={false}
                            to="/passwordreset"
                          >
                            <h6
                              className="pull-right create"
                            >
                              <b>
                                Forgot password?
                              </b>
                            </h6>
                          </Link>
                        </div>
                      </div>
                      <input
                        className="signinform"
                        id="password"
                        name="password"
                        onChange={[Function]}
                        placeholder="*********"
                        required={true}
                        type="password"
                        value=""
                      />
                    </div>
                    <button
                      className="signinformbtn"
                      id="submitButton"
                      type="submit"
                    >
                      Sign in
                    </button>
                  </form>
                </div>
                <br />
                <div>
                  <center>
                    <p>
                      New to PostIt App? 
                      <Link
                        className="create"
                        replace={false}
                        to="/"
                      >
                        Create an account.
                      </Link>
                    </p>
                  </center>
                </div>
              </div>
            </div>
          </div>
          <Footer />
        </div>,
        "_debugID": 893,
        "_renderedOutput": <div>
          <nav
            className="navbar navbar-inverse navabar-fixed-top"
            role="navigation"
          >
            <div
              className="container-fluid"
            >
              <div
                className="navbar-header"
              >
                <button
                  className="navbar-toggle collapsed"
                  data-target=".navbar-collapse"
                  data-toggle="collapse"
                  type="button"
                >
                  <span
                    className="sr-only"
                  >
                    Toggle navigation
                  </span>
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                  <span
                    className="icon-bar"
                  />
                </button>
                <Link
                  className="navbar-brand"
                  replace={false}
                  to="#"
                >
                  PostIt
                  <small>
                    App
                  </small>
                </Link>
              </div>
              <div
                className="collapse navbar-collapse"
              >
                <ul
                  className="nav navbar-nav navbar-right"
                >
                  <li>
                    <Link
                      replace={false}
                      to="/"
                    >
                      Home
                    </Link>
                  </li>
                  <li
                    className="active"
                  >
                    <Link
                      replace={false}
                      to="/signin"
                    >
                      Sign in
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
          <div
            className="container signin"
          >
            <span />
            <div
              className="row"
            >
              <div
                className="col-md-6 col-md-offset-3"
              >
                <div
                  className="row"
                >
                  <center>
                    <GoogleButton
                      disabled={false}
                      label="Sign in with Google"
                      onClick={[Function]}
                      type="dark"
                    />
                  </center>
                  <br />
                  <div
                    className="text-center or"
                  >
                    <b>
                      OR
                    </b>
                  </div>
                  <form
                    className="w3-card w3-white"
                    id="signinForm"
                    onSubmit={[Function]}
                  >
                    <div
                      className="form-group"
                    >
                      <label
                        htmlFor="email"
                      >
                        Email
                      </label>
                      <input
                        className="signinform"
                        id="email"
                        name="email"
                        onChange={[Function]}
                        placeholder="johndoe@example.com"
                        required={true}
                        type="email"
                        value=""
                      />
                    </div>
                    <div
                      className="form-group"
                    >
                      <div
                        className="row"
                      >
                        <div
                          className="col-md-4"
                        >
                          <label
                            htmlFor="password"
                          >
                            Password
                          </label>
                        </div>
                        <div
                          className="col-md-4 col-md-offset-4"
                        >
                          <Link
                            replace={false}
                            to="/passwordreset"
                          >
                            <h6
                              className="pull-right create"
                            >
                              <b>
                                Forgot password?
                              </b>
                            </h6>
                          </Link>
                        </div>
                      </div>
                      <input
                        className="signinform"
                        id="password"
                        name="password"
                        onChange={[Function]}
                        placeholder="*********"
                        required={true}
                        type="password"
                        value=""
                      />
                    </div>
                    <button
                      className="signinformbtn"
                      id="submitButton"
                      type="submit"
                    >
                      Sign in
                    </button>
                  </form>
                </div>
                <br />
                <div>
                  <center>
                    <p>
                      New to PostIt App? 
                      <Link
                        className="create"
                        replace={false}
                        to="/"
                      >
                        Create an account.
                      </Link>
                    </p>
                  </center>
                </div>
              </div>
            </div>
          </div>
          <Footer />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserSignIn />,
}
`;
